Mage/src/main/java/mage/abilities/common/PutCounterOnCreatureTriggeredAbility.java

```diff
@@ -0,0 +1,90 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.counters.Counter;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * "Whenever you put one or more counters on a creature " triggered ability
+ *
+ * @author PurpleCrowbar
+ */
+public class PutCounterOnCreatureTriggeredAbility extends TriggeredAbilityImpl {
+
+    private final Counter counterType; // when null, any counter type is accepted
+    protected FilterPermanent filter;
```

Comment:
Why is this `protected` rather than `private final`?

Reply:
I believe the triggered abilities I was reading to base this off of had their filters as protected (e.g. in the case of **AttackingCreaturePutIntoGraveyardTriggeredAbility** below):

https://github.com/magefree/mage/blob/3a3e3fda48632dd4e2b6933aa637c6ef8634941f/Mage/src/main/java/mage/abilities/common/AttackingCreaturePutIntoGraveyardTriggeredAbility.java#L29

I have tested the new ability with filter as `protected` and `private final` and it seems to function identically. Should it be changed to private final?

Reply:
Yes please. It should be `final` since it should not be changed after being set in the constructor.
The `private` part doesn't make much of a difference atm, only if someone subclassed it. But it can be set to `private` for now, and if needed in a subclass at a late time then it can be changed.